test/message_server/component.adb

```diff
@@ -0,0 +1,71 @@
+
+with Componolit.Gneiss.Message;
+with Componolit.Gneiss.Message.Reader;
+with Componolit.Gneiss.Strings;
+
+package body Component with
+   SPARK_Mode
+is
+
+   type Unsigned_Char is mod 2 ** 8;
+   type C_String is array (Positive range <>) of Unsigned_Char;
+
+   procedure Event;
+
+   package Message_Client is new Gns.Message.Reader (Unsigned_Char, Positive, C_String, 4096, Event);
+
+   type Client_Registry is array (Integer range <>) of Gns.Message.Reader_Session;
+
+   Client  : Client_Registry (1 .. 3);
+   Message : Message_Client.Message_Buffer;
+   M_Str   : String (1 .. 4097);
+
+   procedure Construct (Cap : Gns.Types.Capability)
+   is
+   begin
+      for I in Client'Range loop
+         Message_Client.Initialize (Client (I), Cap, Gns.Strings.Image (I));
+      end loop;
+   end Construct;
+
+   procedure Destruct
+   is
+   begin
+      for I in Client'Range loop
+         Message_Client.Finalize (Client (I));
+      end loop;
+   end Destruct;
+
+   procedure Check_And_Print (I : Integer);
+
+   procedure Check_And_Print (I : Integer)
+   is
+      procedure Print (S1 : in out String;
+                       S2 : in out String) with
+         Import,
+         Convention => C,
+         External_Name => "print_message";
+      Label : String := Gns.Strings.Image (I) & Character'First;
```

Comment:
Why appending `Character'First`? Is that supposed to be a null-terminator? If so, then I'd rather use Character'Val (0) for clarity.

Reply:
Fixed by https://github.com/Componolit/gneiss/pull/117/commits/014bd04f977c5264d663955ec986685d2990debb.